patch-loader/src/main/java/org/lsposed/lspatch/loader/LSPApplication.java

```diff
@@ -243,69 +242,39 @@ private static int getTranscationId(String clsName, String trasncationName) thro
-                        out.readException();
-                        if (0 != out.readInt()) {
-                            PackageInfo packageInfo = PackageInfo.CREATOR.createFromParcel(out);
-                            if (packageInfo.packageName.equals(context.getApplicationInfo().packageName)) {
-                                if (packageInfo.signatures != null && packageInfo.signatures.length > 0) {
-                                    XLog.d(TAG, "Replace signature info (method 1)");
-                                    packageInfo.signatures[0] = new Signature(config.originalSignature);
-                                }
-
-                                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
-                                    if (packageInfo.signingInfo != null) {
-                                        XLog.d(TAG, "Replace signature info (method 2)");
-                                        Signature[] signaturesArray = packageInfo.signingInfo.getApkContentsSigners();
-                                        if (signaturesArray != null && signaturesArray.length > 0) {
-                                            signaturesArray[0] = new Signature(config.originalSignature);
-                                        }
-                                    }
-                                }
-
-                                out.setDataPosition(0);
-                                out.setDataSize(0);
-                                out.writeNoException();
-                                out.writeInt(1);
-                                packageInfo.writeToParcel(out, PARCELABLE_WRITE_RETURN_VALUE);
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                        if (packageInfo.signingInfo != null) {
+                            XLog.d(TAG, "Replace signature info (method 2)");
+                            Signature[] signaturesArray = packageInfo.signingInfo.getApkContentsSigners();
+                            if (signaturesArray != null && signaturesArray.length > 0) {
+                                signaturesArray[0] = new Signature(config.originalSignature);
                             }
                         }
-
-                        // reset pos
-                        out.setDataPosition(0);
                     }
-                } catch (Throwable err) {
-                    // should not happen, just crash app
-                    throw new IllegalStateException("lsp hook error", err);
                 }
+                return packageInfo;
+            }
+
+            @Override
+            public PackageInfo[] newArray(int size) {
+                return originalCreator.newArray(size);
             }
-        });
+        };
+        XposedHelpers.setStaticObjectField(PackageInfo.class, "CREATOR", proxiedCreator);
+        Map<?, ?> mCreators = (Map<?, ?>) XposedHelpers.getStaticObjectField(Parcel.class, "mCreators");
```

Comment:
`sPairedCreators` missing

Reply:
i have tried search `sPairedCreators` in AOSP source tree 4.0 - 13, and find nothing.
is `sPairedCreators` spelled correctly ?

Reply:
https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/os/Parcel.java;l=5118